server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala

```diff
@@ -29,15 +33,28 @@ class TimeoutSpec extends Http4sSpec {
 
     "return the provided response if the result takes too long" in {
       val customTimeout = Response(Status.GatewayTimeout) // some people return 504 here.
-      val altTimeoutService = Timeout(500.millis, Task.now(customTimeout))(myService)
-
+      val altTimeoutService = Timeout(1.millis, Task.now(customTimeout))(myService)
       altTimeoutService.apply(slowReq).run.status must_== (customTimeout.status)
     }
 
-    "Handle infinite durations" in {
+    "handle infinite durations" in {
       val service = Timeout(Duration.Inf)(myService)
       service.apply(slowReq).run.status must_== (Status.Ok)
     }
-  }
 
+    "clean up resources of the loser" in {
```

Comment:
This may be a bit trickier to port to fs2. It's mostly straightforward but playing with external variables is tricky. 

Why are you choosing to use an atomic boolean here, when we use `mutable.Signal` elsewhere for this sort of detection.